### PR TITLE
Update banner interaction logging

### PR DESCRIPTION
**Pull request by Abban**
https://phabricator.wikimedia.org/T250791
https://phabricator.wikimedia.org/T252412

### CODE REVIEW

File: shared/EventLoggingTracker.js [hunk 1]
```diff
@@ -33,9 +36,23 @@ export class EventLoggingTracker {
 				mw.track( 'event.WMDEBannerInteractions', {
 					bannerName: this.bannerName,
 					bannerAction: actionName,
-					bannerImpressions: this.impressionCounter.getImpressionCount()
+					bannerImpressions: this.impressionCounter.getImpressionCount(),
+					userId: mw.user.id
```
Comment (KaiNissen) [inline shared/EventLoggingTracker.js]: I think, `mw.user.getId()` is what we want.

The schema defines this field as `userID` (capital D).

File: shared/EventLoggingTracker.js [hunk 2]
```diff
@@ -33,9 +36,23 @@ export class EventLoggingTracker {
...
+	trackSeenEvent( trackingRatio ) {
+		if ( Math.random() >= trackingRatio ) {
+			return;
+		}
+
+		mw.track( 'event.WMDEBannerInteractions', {
+			bannerName: this.bannerName,
+			bannerAction: 'banner-seen',
+			bannerImpressions: this.impressionCounter.getImpressionCount(),
+			userId: mw.user.id
```
Comment (KaiNissen) [inline shared/EventLoggingTracker.js]: Same here.

File: occasional_editors/Banner.js [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+const BANNER_CONTAINER_ID = 'WMDE-Banner-Container';
+const VECTOR_PANEL = 'mw-panel';
+const VECTOR_HEAD = 'mw-head';
+const MONOBOOK_WRAPPER = 'globalWrapper';
+const CENTRAL_NOTICE_ID = 'centralNotice';
+const EDIT_BUTTONS_CLASS = 'mw-editsection-visualeditor';
+const EDIT_BUTTON_ID = 'ca-ve-edit';
+
+const BANNER_CLOSE_TRACK_RATIO = 0.01;
+const BANNER_SEEN_TRACK_RATIO = 0.01;
```
Comment (KaiNissen) [inline occasional_editors/Banner.js]: Banners are shown to a very limited group of users, so we should increase this to `1`.

**Review by KaiNissen — CHANGES_REQUESTED** [review]
Some minor changes required to make this work.

File: shared/EventLoggingTracker.js [hunk 4]
```diff
@@ -33,9 +36,23 @@ export class EventLoggingTracker {
...
-					bannerImpressions: this.impressionCounter.getImpressionCount()
+					bannerImpressions: this.impressionCounter.getImpressionCount(),
+					userId: mw.user.id
 				} );
 			}
 		};
 	}
+
+	trackSeenEvent( trackingRatio ) {
+		if ( Math.random() >= trackingRatio ) {
```
Comment (KaiNissen) [inline shared/EventLoggingTracker.js]: Shouldn't this be `<=`?

**Review by gbirke — APPROVED** [review]

**Pull request merged**